test/test_codegen_fortran.py

```diff
@@ -150,19 +150,35 @@ def test_self_dep_in_loop():
         fortran_libraries=["lapack", "blas"])
 
 
+class AbsFailure:
+    pass
+
+
 def test_elementwise_abs():
     with CodeBuilder(name="primary") as cb:
         cb("y", "<func>f(0, <state>ytype)")
         cb("<state>ytype", "y")
         # Test new builtin on a usertype.
         cb("z", "<builtin>elementwise_abs(<state>ytype)")
+        with cb.if_("z[1] > 2"):
```

Comment:
It looks like this is indexing into a user type, which wouldn't be allowed. Can you help me understand? If you'd like to assert something about the internal state of a user type, it should be done in Fortran.

Reply:
Yeah, this is a fair point. Obviously given the inadequacy of the previous version of the test I wanted to make sure I'm checking the result when the function is applied to scalars, arrays, and usertypes, but without indexing into the UserType (which is just an array) I'm not sure how to check it. Can you explain what you mean by the last part of your comment?

Reply:
Nvm - I *think* I did what makes more sense in 0237e4d. (Let me know if this isn't what you meant)